### PR TITLE
Add more unphysical cell checks for MHM prediction.

### PR DESCRIPTION
**Pull request by ChunYen-Chen**
As title. We also changed the default value of  `AUTO_REDUCE_DT_FACTOR` to `1.0` in this PR.

### CODE REVIEW

File: example/test_problem/Hydro/CMZ/Input__Parameter [hunk 1]
```diff
@@ -81,7 +81,7 @@ OPT__DT_USER                  0           # dt criterion: user-defined -> edit "
 OPT__DT_LEVEL                 3           # dt at different AMR levels (1=shared, 2=differ by two, 3=flexible) [3]
 OPT__RECORD_DT                1           # record info of the dt determination [1]
 AUTO_REDUCE_DT                0           # reduce dt automatically when the program fails (for OPT__DT_LEVEL==3 only) [1]
-AUTO_REDUCE_DT_FACTOR         0.8         # reduce dt by a factor of AUTO_REDUCE_DT_FACTOR when the program fails [0.8]
+AUTO_REDUCE_DT_FACTOR         1.0        # reduce dt by a factor of AUTO_REDUCE_DT_FACTOR when the program fails [1.0]
```
Comment (hyschive) [inline example/test_problem/Hydro/CMZ/Input__Parameter:84]: Align the code

File: src/Model_Hydro/CPU_Hydro/CPU_Shared_DataReconstruction.cpp [hunk 2]
```diff
@@ -1974,42 +1974,46 @@ void Hydro_HancockPredict( real fc[][NCOMP_LR], const real dt, const real dh,
...
+#     ifndef BAROTROPIC_EOS
+      if ( fc[f][4] <= (real)0.0 || fc[f][4] >= HUGE_NUMBER || fc[f][4] != fc[f][0] ) reset_cell = true;
+      if ( Pres     <= (real)0.0 || Pres     >= HUGE_NUMBER || Pres     != fc[f][0] ) reset_cell = true;
+#     endif
+
+//    set to the cell-centered values before update
+      if ( reset_cell )
       {
-//       set to the cell-centered values before update
          for (int f=0; f<6; f++)
```
Comment (hyschive) [inline src/Model_Hydro/CPU_Hydro/CPU_Shared_DataReconstruction.cpp:1999]: The variable `f` has been used in the outer loop. Rename it for clarity.

File: src/Model_Hydro/CPU_Hydro/CPU_Shared_DataReconstruction.cpp [hunk 3]
```diff
@@ -1974,42 +1974,46 @@ void Hydro_HancockPredict( real fc[][NCOMP_LR], const real dt, const real dh,
    MHD_UpdateMagnetic_Half( fc, g_EC_Ele, dt, dh, cc_i-NGhost, cc_j-NGhost, cc_k-NGhost, NEle );
 #  endif
 
-// check the negative density and energy
+   bool reset_cell = false;
    for (int f=0; f<6; f++)
    {
-#     ifdef BAROTROPIC_EOS
-      if ( fc[f][0] <= (real)0.0 )
+#     ifdef MHD
```
Comment (hyschive) [inline src/Model_Hydro/CPU_Hydro/CPU_Shared_DataReconstruction.cpp:1980]: Compute `Emag` and `Pres` inside `#ifndef BAROTROPIC_EOS`

File: src/Model_Hydro/CPU_Hydro/CPU_Shared_DataReconstruction.cpp [hunk 4]
```diff
@@ -1974,42 +1974,46 @@ void Hydro_HancockPredict( real fc[][NCOMP_LR], const real dt, const real dh,
...
 #     endif
+      const real Pres = Hydro_Con2Pres( fc[f][DENS], fc[f][MOMX], fc[f][MOMY], fc[f][MOMZ], fc[f][ENGY], fc[f]+NCOMP_FLUID,
+                                        true, MinPres, Emag, EoS->DensEint2Pres_FuncPtr, EoS->AuxArrayDevPtr_Flt,
+                                        EoS->AuxArrayDevPtr_Int, EoS->Table, NULL );
+
+//    check the negative, inf or nan density, energy, and pressure
+      if ( fc[f][0] <= (real)0.0 || fc[f][0] >= HUGE_NUMBER || fc[f][0] != fc[f][0] ) reset_cell = true;
+#     ifndef BAROTROPIC_EOS
+      if ( fc[f][4] <= (real)0.0 || fc[f][4] >= HUGE_NUMBER || fc[f][4] != fc[f][0] ) reset_cell = true;
+      if ( Pres     <= (real)0.0 || Pres     >= HUGE_NUMBER || Pres     != fc[f][0] ) reset_cell = true;
```
Comment (hyschive) [inline src/Model_Hydro/CPU_Hydro/CPU_Shared_DataReconstruction.cpp:1993]: Shouldn't it be `Pres != Pres`?

File: src/Model_Hydro/CPU_Hydro/CPU_Shared_DataReconstruction.cpp [hunk 5]
```diff
@@ -1974,42 +1974,46 @@ void Hydro_HancockPredict( real fc[][NCOMP_LR], const real dt, const real dh,
...
+//    set to the cell-centered values before update
+      if ( reset_cell )
       {
-//       set to the cell-centered values before update
          for (int f=0; f<6; f++)
          for (int v=0; v<NCOMP_TOTAL; v++)
             fc[f][v] = g_cc_array[v][cc_idx];
 
-         break;
+         break;  // no need to apply the floors since the input values should already applied
```
Comment (hyschive) [inline src/Model_Hydro/CPU_Hydro/CPU_Shared_DataReconstruction.cpp:2003]: "applied" --> "satisfy these constraints"

File: src/Model_Hydro/CPU_Hydro/CPU_Shared_DataReconstruction.cpp [hunk 6]
```diff
@@ -1974,42 +1974,46 @@ void Hydro_HancockPredict( real fc[][NCOMP_LR], const real dt, const real dh,
...
-#     endif // #ifndef BAROTROPIC_EOS
 #     if ( NCOMP_PASSIVE > 0 )
-      for (int v=NCOMP_FLUID; v<NCOMP_TOTAL; v++)
-      fc[f][v] = FMAX( fc[f][v], TINY_NUMBER );
+      for (int v=NCOMP_FLUID; v<NCOMP_TOTAL; v++) fc[f][v] = FMAX( fc[f][v], TINY_NUMBER );
 #     endif
-   }
+
+   } // for (int f=0; f<6; f++)
+
```
Comment (hyschive) [inline src/Model_Hydro/CPU_Hydro/CPU_Shared_DataReconstruction.cpp:2016]: Remove this empty line.

**Review by hyschive — COMMENTED** [review]
@ChunYen-Chen Good. I've added some minor comments.

File: src/Model_Hydro/CPU_Hydro/CPU_Shared_DataReconstruction.cpp (hunk 4 above)
Comment (ChunYen-Chen) [reply src/Model_Hydro/CPU_Hydro/CPU_Shared_DataReconstruction.cpp]: Yes. I also made a mistake at the previous line. I have fixed it.

**Comment by ChunYen-Chen** [comment]
@hyschive Thanks for reviewing. Changes have been made.

**Review by hyschive — APPROVED** [review]

**Pull request merged**